src/Network/Transport/TCP/Internal.hs

```diff
@@ -90,22 +112,38 @@ forkServer host port backlog reuseAddr terminationHandler requestHandler = do
       when reuseAddr $ N.setSocketOption sock N.ReuseAddr 1
       N.bindSocket sock (N.addrAddress addr)
       N.listen sock backlog
+
+      -- Close up and fill the synchonizing MVar.
+      let release :: ((N.Socket, N.SockAddr), MVar ()) -> IO ()
+          release ((sock, _), socketClosed) =
+            N.sClose sock `finally` putMVar socketClosed ()
+
+      -- Accept connections and handle them in separate threads, taking care
+      -- to close the socket when its handler is finished.
+      let acceptRequest :: IO ()
+          acceptRequest = do
+            (sock, sockAddr) <- N.accept sock
```

Comment:
Can this thread be interrupted? sock could leak if so. `accept` happened in a `bracket` call before.